### PR TITLE
add normalizer for am_i_eligible_covid19vaccine_gov

### LINKED ISSUE #68: normalize am-i-eligible.covid19vaccine.health.ny.gov from New York
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Blocked by #67 

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch ny/am_i_eligible_covid19vaccine_gov
    poetry run vaccine-feed-ingest parse ny/am_i_eligible_covid19vaccine_gov
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize ny/am_i_eligible_covid19vaccine_gov
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by rascalking**
also updates the parser to insert the feed-level `lastUpdated` field into each site.

this should resolve #68.

### CODE REVIEW

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+NAME_CLEAN_RE = re.compile("^[\u1d42*]+")
+
+
+def normalize(site_blob: dict, timestamp: str) -> dict:
+    """
+    sample entry:
+
+    {"providerId": 1013, "providerName": "\u1d42**York College - Health and Physical Education Complex - Queens", "vaccineBrand": "Pfizer", "address": "Jamaica, NY", "availableAppointments": "Y", "isShowable": true, "lastUpdated": "2021-04-23T20:04:24"} # noqa: E501
+    """
+    name = NAME_CLEAN_RE.sub("", site_blob["providerName"]).strip()
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py:23]: It looks to me like the normalized output has a few of the leading `\u1d42`s left in, e.g., `"providerName": "\u1d42Suffolk CCC - Brentwood",`.

Should we consider a regex more like `^(\\u1d42)?\**`?

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/parse.py [hunk 2]
```diff
@@ -14,8 +15,13 @@
 with input_file.open() as fin:
     data = json.load(fin)
 
+# TODO: if we want to pull in pytz, we can add in the eastern tz here
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/parse.py:18]: If we need it - feel free to add it (`poetry add <foo>`)! I will defer to @rajbot or @bryanculbertson on time-parsing advice in Python.

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "address": {
+            "city": city,
+            "state": "NY",
+        },
+        "availability": {
+            "appointments": appts_available,
+        },
+        "inventory": [
+            {
+                "vaccine": brand,
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py:40]: CC: @rajbot / @bryanculbertson - do we have a defined enum for what these should normalize to? If not... let's add one!

No need to make any changes in this PR, we can address as a followup once the data schema is solidified.

**Review by eliblock — COMMENTED** [review]
Thank you for adding this normalizer (on top of the fetcher and parser you added earlier today)! It looks like we might need to tweak the name cleaning a bit further, but this is almost good to go!

For future reference: we have a (typed) schema which you can use instead of hand-constructing the json - see an example in [`ak/arcgis/normalize.py`](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/af8d335d45a674caf7906adcce2de621e695f63e/vaccine_feed_ingest/runners/ak/arcgis/normalize.py#L124-L156). No need to change anything here if you don't want to, but it might make writing future normalizers a bit easier (you'll get type hints, etc.).

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/parse.py (hunk 2 above)
Comment (rascalking) [reply vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/parse.py:18]: Oh, cool, just didn't want to pull in new dependencies willy-nilly.  Adding that in.

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py (hunk 1 above)
Comment (rascalking) [reply vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py:23]: I'm actually not seeing that on my end.

```
ono-sendai:runners bonnerd$ rm -rf out
ono-sendai:runners bonnerd$ poetry run vaccine-feed-ingest fetch ny/am_i_eligible_covid19vaccine_gov
ono-sendai:runners bonnerd$ poetry run vaccine-feed-ingest parse ny/am_i_eligible_covid19vaccine_gov
ono-sendai:runners bonnerd$ poetry run vaccine-feed-ingest normalize ny/am_i_eligible_covid19vaccine_gov
ono-sendai:runners bonnerd$ cat out/ny/am_i_eligible_covid19vaccine_gov/normalized/2021-04-24T11\:07\:02/data.normalized.ndjson | jq '.name' | rg -i brentwood
"Suffolk CCC - Brentwood"
```

File: vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py (hunk 3 above)
Comment (rascalking) [reply vaccine_feed_ingest/runners/ny/am_i_eligible_covid19vaccine_gov/normalize.py:40]: Yup, not a problem.

**Comment by rascalking** [comment]
> For future reference: we have a (typed) schema which you can use instead of hand-constructing the json - see an example in [`ak/arcgis/normalize.py`](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/af8d335d45a674caf7906adcce2de621e695f63e/vaccine_feed_ingest/runners/ak/arcgis/normalize.py#L124-L156). No need to change anything here if you don't want to, but it might make writing future normalizers a bit easier (you'll get type hints, etc.).

Gotcha.  I was working off the `vaccinespotter_org` normalizer as a jumping off point, but happy to switch it over.

**Comment by rascalking** [comment]
Turns out the `schema.Address` type won't validate without `street1` and `zip`, which this feed doesn't supply, so we're using the schema for most of it, then converting to a dict and inserting the address then.

**Review by eliblock — APPROVED** [review]

**Comment by eliblock** [comment]
@rascalking thank you for getting this data fetched, parsed, and normalized so quickly!

**Pull request merged**